agent/structs/config_entry_discoverychain.go

```diff
@@ -0,0 +1,746 @@
+package structs
```

Comment:
I split out these 3 related config entry types to keep the interrelated types separate from other config entries.